Sources/Keychain/secretkey/KeyEncrypting.swift

```diff
@@ -120,4 +123,30 @@ public final class KeyEncrypting: KeyEncryptable {
 
         return result
     }
+
+    /// Fetch number of secrets using address as reference
+    private func getSecretCount(with reference: String) -> Int {
+        let query: [String: Any] = [
+            kSecClass as String: kSecClassKey,
+            kSecAttrKeyType as String: kSecAttrKeyTypeECSECPrimeRandom,
+            kSecAttrApplicationTag as String: reference.data(using: .utf8) as Any,
+            kSecReturnRef as String: true,
+            kSecMatchLimit as String: kSecMatchLimitAll
+        ]
+
+        // 1. SecItemCopyMatching will query how many secrets are currently stored in the secure enclave
+        var secretRef: CFTypeRef?
+        let status = security.SecItemCopyMatching(
+            query as CFDictionary,
+            &secretRef
+        )
+
+        // 2. Return how many secrets were retrieved from the secure enclave
+        if status == noErr, let secrets = secretRef as? [SecKey] {
+            return secrets.count
```

Comment:
Why not always just return here? I think this is saying if there's no error, then return the number of secrets. 

Why also have `return 0` on line 150?

Reply:
the keychain returns an optional value, secretRef in our case - we return 0 if our output cannot be cast to an array

Reply:
Okay. In TypeScript/Python you'd check if `secretRef` was undefined in that case and return `0` if it wasn't. Maybe a different standard in Swift - this works too